### ISSUE TITLE
recent removing of call to set_status_and_headers_in_response breaks application

### ISSUE THREAD

**Issue by marksantcroos** [marshmallow-code/flask-smorest#176]
https://github.com/marshmallow-code/flask-smorest/commit/2a793adf27cbbe41f5940ccc115bfd153cb0c41c breaks my application, although I find it difficult to pinpoint why, as my integration tests fail with a database error.

I had to do a binary search to find the commit as it was not related to the mentioned API changes between 0.23.0 and 0.24.0.

Anything I can do to assist debugging?

**Comment by marksantcroos** [comment]
I traced it to a `POST` that now returns `200` instead of `202` which I would expect.

**Comment by lafrech** [comment]
Thanks for reporting.

I was sure this call was useless. I guess I need to rethink this.

Currently OOO with limited internet access. I'm putting this on top of my TODO-list and I'll try to release a bugfix asap.

A resource func example might help craft a test case.

**Comment by marksantcroos** [comment]
Thanks for your quick response. Three files inlined below, request `pytest`.

With the old behavior, 2 out of ABC succeed, with the new behavior, 1 out of ABC succeed. Can you comment on the proper usage of jsonify and the response code. Or shouldn't I be using jsonify at all?

Thanks!


The application:
``` python
from flask import Flask, jsonify
from flask_smorest import Api, Blueprint


app = Flask(__name__)
app.config['API_TITLE'] = 'My API'
app.config['API_VERSION'] = 'v1'
app.config['OPENAPI_VERSION'] = '3.0.2'
api = Api(app)

blp = Blueprint('test', 'test')

@blp.route("/A", methods=["POST"])
@blp.response(code=202)
def post():
    return jsonify({"message": "202"}), 202


@blp.route("/B", methods=["POST"])
def post():
    return jsonify({"message": "202"}), 202


@blp.route("/C", methods=["POST"])
@blp.response(code=202)
def post():
    return jsonify({"message": "202"})


api.register_blueprint(blp)
```


`conftest.py`:
``` python
import pytest


@pytest.fixture
def client():
    from myapp import app
    return app.test_client()
```

`test_abc.py`:
``` python
def test_index_post_A(client):
    response = client.post("/A")
    print(response)
    assert response.status_code == 202


def test_index_post_B(client):
    response = client.post("/B")
    print(response)
    assert response.status_code == 202


def test_index_post_C(client):
    response = client.post("/C")
    print(response)
    assert response.status_code == 202
```

**Comment by lafrech** [comment]
Fix released in 0.24.1.
